cs-office-365/src/main/java/io/cloudslang/content/office365/actions/utils/CreateMessage.java

```diff
@@ -0,0 +1,216 @@
+
+import com.hp.oo.sdk.content.annotations.Action;
+import com.hp.oo.sdk.content.annotations.Output;
+import com.hp.oo.sdk.content.annotations.Param;
+import com.hp.oo.sdk.content.annotations.Response;
+import io.cloudslang.content.constants.ReturnCodes;
+import io.cloudslang.content.office365.entities.CreateMessageInputs;
+import io.cloudslang.content.office365.entities.Office365CommonInputs;
+import io.cloudslang.content.office365.services.EmailServiceImpl;
+import io.cloudslang.content.utils.StringUtilities;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.hp.oo.sdk.content.plugin.ActionMetadata.MatchType.COMPARE_EQUAL;
+import static com.hp.oo.sdk.content.plugin.ActionMetadata.ResponseType.ERROR;
+import static com.hp.oo.sdk.content.plugin.ActionMetadata.ResponseType.RESOLVED;
+import static io.cloudslang.content.constants.OutputNames.*;
+import static io.cloudslang.content.constants.ResponseNames.FAILURE;
+import static io.cloudslang.content.constants.ResponseNames.SUCCESS;
+import static io.cloudslang.content.httpclient.entities.HttpClientInputs.*;
+import static io.cloudslang.content.office365.services.EmailServiceImpl.createMessage;
+import static io.cloudslang.content.office365.utils.Constants.*;
+import static io.cloudslang.content.office365.utils.Descriptions.Common.*;
+import static io.cloudslang.content.office365.utils.Descriptions.CreateMessage.*;
+import static io.cloudslang.content.office365.utils.Descriptions.GetAuthorizationToken.FAILURE_DESC;
+import static io.cloudslang.content.office365.utils.Descriptions.GetAuthorizationToken.SUCCESS_DESC;
+import static io.cloudslang.content.office365.utils.Inputs.CommonInputs.PROXY_HOST;
+import static io.cloudslang.content.office365.utils.Inputs.CommonInputs.PROXY_PASSWORD;
+import static io.cloudslang.content.office365.utils.Inputs.CommonInputs.PROXY_PORT;
+import static io.cloudslang.content.office365.utils.Inputs.CommonInputs.PROXY_USERNAME;
+import static io.cloudslang.content.office365.utils.Inputs.CreateMessage.*;
+import static io.cloudslang.content.office365.utils.Inputs.CreateMessage.BODY;
+import static io.cloudslang.content.office365.utils.Inputs.EmailInputs.*;
+import static io.cloudslang.content.office365.utils.InputsValidation.verifyCreateMessageInputs;
+import static io.cloudslang.content.office365.utils.Outputs.CommonOutputs.DOCUMENT;
+import static io.cloudslang.content.utils.OutputUtilities.getFailureResultsMap;
+import static io.cloudslang.content.utils.OutputUtilities.getSuccessResultsMap;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+
+public class CreateMessage {
+
+    /**
+     * @param userPrincipalName          The email address of the user to perform the action on. The input is mutually exclusive
+     *                                   with the 'userId' input.
+     * @param userId                     The ID of the user to perform the action on.
```

Comment:
Shouldn't you add: "The input is mutually exclusive with the 'userPrincipalName' input." ??